e2e-test/src/tests/delete.test.js

```diff
@@ -0,0 +1,66 @@
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+
+    // click on 'X'
+    await page.waitForXPath("//div[@class='v-btn__content']//following::i[@class='v-icon mdi mdi-close theme--light']");
+    await expect(page).toClickXPath(vBtn("//i[@class='v-icon mdi mdi-close theme--light']"));
+
+    // make sure it is deleting one dataset
+    await page.waitForXPath("//div[@class='v-card__title headline'][contains(text(),'Really delete 1 dataset?')]");
+    await expect(page).toClickXPath(vBtn('Delete'));
+
+    // make sure it is deleted
+    await expect(page).not.toContainXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+  });
+
+  it('upload 3 datasets and then clear all', async () => {
+    await Promise.all([
+      page.goto(CLIENT_URL),
+      page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+
+    const fileInputElementHandle = await page.waitForXPath("//input[@type='file']");
+
+    // upload dataset 1
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+
+    // upload dataset 2
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo liver (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_liver_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_liver_NMR.csv ')]//following::span[contains(text(),'Dataset ready for analysis')]");
+
+    // upload dataset 3
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo serum (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_serum_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_serum_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+
+    // click on 'clear all'
+    await expect(page).toClickXPath(vBtn('clear all'));
+
+    // make sure deleting 3 at the same time
+    await page.waitForXPath("//div[@class='v-card__title headline'][contains(text(),'Really delete 3 datasets?')]");
```

Comment:
```suggestion
    await page.waitForXPath(vCard({ title: 'Really delete 3 datasets?' }));
```

Reply:
Also, if there are any leftover datasets, clearing all might give a number larger than 3. This test should probably be clicking this button at the beginning of the test to make sure there are no extra files hanging around to interfere.

In fact, that might be a handy utility function that could be used in any tests that don't want extra files hanging around. What do you think?

Reply:
Yeah I agree. So write a function an reuse it for different tests to make sure all the files are cleared?

Reply:
Yep, specifically for tests that need to start with no existing files (like this one). Most tests probably won't use it.